src/libraries/Microsoft.PowerFx.Core/Public/Marshal/ITypeMashallerProvider.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx.Core
+{
+    /// <summary>
+    /// Handles marshalling a given type. Invoked by the <see cref="TypeMarshallerCache"/>.
+    /// </summary>
+    public interface ITypeMashallerProvider
+    {
+        // Return null if it doesn't handle it. 
+        // A single ITypeMarshaler can be created once per type and then reused for each instance.
```

Comment:
nit: for these comments on public interfaces, use 
/// <summary> style doc comments so it gets picked up by intellisense in hosts.